packages/playground/src/components/node_details.vue

```diff
@@ -0,0 +1,176 @@
+import toTeraOrGigaOrPeta from "../utils/toTeraOrGegaOrPeta";
+
+const dNodeError = ref(false);
+const dNodeLoading = ref(true);
+const farmName = ref("");
+const publicIps = ref(0);
+const NodeResourceFields = ref<NodeDetailsCard[]>();
+const CountryResourceFields = ref<NodeDetailsCard[]>();
+const FarmResourceFields = ref<NodeDetailsCard[]>();
+const node = ref<GridNode>(nodeInitializer);
+
+const props = defineProps({
+  item: {
+    required: true,
+    type: null,
+  },
+  columnsLen: {
+    required: true,
+    type: Number,
+  },
+});
+
+onMounted(async () => {
+  getNodeDetails();
+});
+
+async function getNodeDetails() {
+  try {
+    dNodeLoading.value = true;
+    const res = await gridProxyClient.farms.list({ farmId: props.item.value.farmId });
+    farmName.value = res.data[0].name;
+    publicIps.value = res.data[0].publicIps.length;
+    if (Array.isArray(res) && !res.length) throw new Error("Can't resolve farm data");
+    NodeResourceFields.value = getNodeResourceCard();
+    CountryResourceFields.value = getCountryResourceCard();
+    FarmResourceFields.value = getFarmResourceCard();
+    if (props.item.value.num_gpu > 0) {
+      const _node: GridNode = await getNode(props.item.value.nodeId, {
+        loadGpu: true,
+      });
+      node.value = _node;
+    }
+    dNodeError.value = false;
+  } catch (e) {
+    dNodeError.value = true;
+    dNodeLoading.value = false;
+  }
+  dNodeLoading.value = false;
+}
+
+const getFarmResourceCard = (): NodeDetailsCard[] => {
+  return [
+    { name: "ID", value: props.item.value.farmId },
+    { name: "Name", value: farmName.value },
+    { name: "Certification type", value: props.item.value.certificationType },
+    { name: "Public ips", value: publicIps.value },
+  ];
+};
+
+const getCountryResourceCard = (): NodeDetailsCard[] => {
+  return [
+    { name: "Country", value: props.item.value.location.country },
+    { name: "City", value: props.item.value.location.city },
+    { name: "Latitude", value: props.item.value.location.latitude },
+    { name: "Longitude", value: props.item.value.location.longitude },
+  ];
+};
+
+const getNodeResourceCard = (): NodeDetailsCard[] => {
+  return [
+    { name: "CPU", value: props.item.value.total_resources.cru + " CPU" },
+    { name: "Memory", value: toTeraOrGigaOrPeta(props.item.value.total_resources.mru) },
+    { name: "Disk(SSD)", value: toTeraOrGigaOrPeta(props.item.value.total_resources.sru) },
+    { name: "Disk(HDD)", value: toTeraOrGigaOrPeta(props.item.value.total_resources.hru) },
+  ];
+};
+
+function getColSize() {
+  if (props.item.num_gpu > 0) {
+    return 6;
+  } else {
+    return 4;
+  }
+}
+</script>
+
+<script lang="ts">
+import CardDetails from "@/explorer/components/node_details_cards/card_details.vue";
+import GPUDetailsCard from "@/explorer/components/node_details_cards/gpu_details_card.vue";
+
+export default {
+  name: "Node Details",
+  components: {
+    CardDetails,
+    GPUDetailsCard,
+  },
+};
```

Comment:
I suggest moving `node_details` to `portal/components/node_details.vue` for clarity. Placing it in the main component folder might cause confusion, especially for new developers. This way, it's easier for everyone to understand its purpose.

All Portal related component can be moved IMHO